sklearn/metrics/tests/test_regression.py

```diff
@@ -19,6 +20,8 @@
 
 from sklearn.metrics.regression import _check_reg_targets
 
+from ...exceptions import DataDimensionalityWarning
```

Comment:
Wouldn't `UndefinedMetricWarning` be more apt? Or even a simple `UserWarning`. What do you think @jnothman ?

Reply:
You're right, `UndefinedMetricWarning` makes more sense with that warning message and is more consistent with other warnings. Let's see what @jnothman has to say.